tools/generators/lib/PBXProj/src/Identifiers.swift

```diff
@@ -6,45 +6,74 @@ import CryptoKit
 /// Since the various `PBXProj` partial generators need to work independently
 /// from each other, they need help to be able to generate unique identifiers.
 /// `Identifiers` is used to generate these identifiers.
+///
+/// The identifiers aren't all hashes. Grouping information is also encoded into
+/// them. The first two characters indicate that it's a global element (`FF`), a
+/// file or group element (`FE`), or a generator shard (`00`-`FD`).
+///
+/// - Global elements (which start with `FF`) are then classified as belonging
+///   to the project as a whole (`00`) or the `BazelDependencies` target
+///   (`01`). For both of them, their associated elements (e.g. configuration
+///   list, build phases, etc.) use `000000000000000000`-`0000000000000000FF`,
+///   and their configurations (e.g. "Debug", "Release", etc.) use
+///   `000000000000000100`-`0000000000000001FF`.
```

Comment:
Are these missing two digits? Seems the partial tests identifiers are 24 characters and these with `FF` & `00`/`01` are only 22 characters.

Reply:
Correct, they are off by `00`. Will fix.